test/functional/gridfs_stream.test.js

```diff
@@ -1380,4 +1382,44 @@ describe('GridFS Stream', function() {
       // END
     }
   });
+
+  it('should correctly handle indexes create with number', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const db = client.db(configuration.db);
+      const col = db.collection('fs.files');
+      col.createIndex({ filename: 1, uploadDate: 1 }, err => {
+        expect(err).to.not.exist;
+        col.listIndexes().toArray((err, indexes) => {
+          expect(err).to.not.exist;
+          const names = indexes.map(i => i.name);
+          expect(names).to.eql(['_id_', 'filename_1_uploadDate_1']);
+          client.close();
+          done();
+        });
+      });
+    });
+  });
+
+  it('should correctly handle indexes create with decimal', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const db = client.db(configuration.db);
+      const col = db.collection('fs.files');
+      col.createIndex({ filename: 1.0, uploadDate: 1.0 }, err => {
```

Comment:
I'm not sure of two things:

1) Are we simply trying to create indexes with the driver using floats? If so I'm a little confused what this [has to do with the shell.](https://jira.mongodb.org/browse/DRIVERS-789) It was my understanding the type mis-match would occur if the shell creates the index, and the driver loads the collection with that created index. Just need clarification on what we're looking for here.

2) What makes a collection in the Node Driver a GridFs collection? There's nothing specific to these tests that differentiates them from creating an index on a regular collection.

Reply:
Not sure, but I think:
1. We don't need a test at all, because both the Mongo shell and the Node driver are using JS and these sorts of differences aren't an issue, because `1 === 1.0` in JS.
2. GridFS manages two collections, `<bucket>.files` and `<bucket>.chunks` for each bucket (in this case `fs`). You can access these just like any other collection, which is why you can create an index on them using `Collection.createIndex`, but they are also special because you can access them through GridFS.

Reply:
@emadum thanks for clarifing